pixi/src/ui/PageExperience.js

```diff
@@ -42,17 +48,42 @@ export default class PageExperience {
     } else {
       this.toggleLoading(true);
 
-      const check = new PageExperienceCheck();
-      const report = await check.run(inputUrl);
-
-      for (const [id, metric] of Object.entries(
-        report.coreWebVitals.fieldData
-      )) {
-        this.reportViews[id] =
-          this.reportViews[id] || new CoreWebVitalsReport(document, id);
-        this.reportViews[id].render(metric);
-      }
-      // TODO: Show error message in UI
+      // Gather errors from all test runs
+      const errors = [];
+
+      // Core Web Vitals
+      this.pageExperienceCheck.run(inputUrl).then((pageExperienceReport) => {
```

Comment:
use async await

Reply:
How would this be rewritten using `async`/`await` without making each check blocking? The only way I can think of would be to move this to a separate method and use it there:

```js
onSubmitUrl(e) {
  // Nor await or then here
  this.runPageExperienceCheck(e.value);
...
}

async runPageExperienceCheck(inputUrl) {
  const {error, data} = await this.pageExperienceCheck.run(inputUrl);
  if(error) {
 ...
}
```

Is this a structure you'd prefer or am I missing something?

Reply:
Sorry, @caroqliu was so kind to point me to the relavant section in the design dok. So your idea was the following:

```js
const pageExperienceCheckRun = this.pageExperienceCheck.run(inputUrl);
const safeBrowsingCheckRun = this.safeBrowsingCheck.run(inputUrl);

const pageExperienceCheckResult = await pageExperienceCheckRun;
const safeBrowsingCheckResult = await safeBrowsingCheckRun;

// Render all results
```

This way we could only render all results at once. I think it would be quite nice if each check pops up as soon as it has run. WDYT?

Reply:
Yup, separate method would good (anyway). Maybe:

```
const checksPromise = Promise.all([
  runCheck(url),
  ...
])
```